Setup.sh - `src/deps/libmimalloc.a`: No such file or directory

### What version of Bun is running?

1.0.22+b400b36ca

### What platform is your computer?

Linux 6.2.0-1018-azure x86_64 x86_64

### What steps can reproduce the bug?

1. `sudo apt install cargo ccache cmake git golang libtool ninja-build pkg-config rustc ruby-full xz-utils`
2. `curl -fsSL https://bun.sh/install | bash`
3. `wget https://apt.llvm.org/llvm.sh -O - | sudo bash -s -- 16 all`
4. `bun setup`

### What is the expected behavior?

Bun is built, without any errors, including `libmimalloc-debug`.

### What do you see instead?

Bun attempts to link `libmimalloc-valgrind-debug.a`, but fails because `src/deps/libmimalloc.a` doesn't exist:

```sh
-- Build files have been written to: /workspaces/bun/src/deps/mimalloc
+ ninja
[17/17] Linking CXX static library libmimalloc-valgrind-debug.a
+ cp libmimalloc-debug.a /workspaces/bun/src/deps//libmimalloc-debug.a
cp: cannot stat 'libmimalloc-debug.a': No such file or directory
Failed to build mimalloc-debug
error: script "setup" exited with code 1
```

Interestingly, `src/deps/libmimalloc` doesn't show up in VSCode either, does that have to do with something? It makes sense because of the double slash, but I'm not sure why it then claims that build files have been written.

### Additional information

Running on GitHub Codespaces. I used `sudo apt update` and `sudo apt upgrade` before running.

After digging a bit last night I thought that maybe it's the `CMake` which isn't configured correctly. 
```bash
devansh   3297_stream_writable  cmake --version
CMake Error: Could not find CMAKE_ROOT !!!
CMake has most likely not been installed correctly.
Modules directory not found in
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.28.1/share/cmake-3.22
cmake version 3.22.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).

```
So, I again installed everything correctly but still I am facing the same probem -

```bash
 [~/bun]
 ✘  devansh   3297_stream_writable  cmake --version
cmake version 3.28.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
               
 [~/bun]
 devansh   3297_stream_writable  bun setup              
$ ./scripts/setup.sh
All system dependencies OK
C Compiler for dependencies: /usr/bin/clang-16
C++ Compiler for dependencies: /usr/bin/clang++-16
+ git submodule update --init --recursive --progress --depth=1 --checkout src/deps/picohttpparser src/deps/mimalloc src/deps/zlib src/deps/libarchive src/deps/boringssl src/deps/lol-html src/deps/tinycc src/deps/c-ares src/deps/zstd src/deps/base64 src/deps/ls-hpack
base64 - already built
boringssl - already built
cares - already built
libarchive - already built
lolhtml - already built
building mimalloc-debug
++ dirname -- /home/devansh/bun/scripts/build-mimalloc-debug.sh
+ source /home/devansh/bun/scripts/env.sh
++++ dirname -- /home/devansh/bun/scripts/env.sh
+++ cd -- /home/devansh/bun/scripts
+++ pwd
++ export SCRIPT_DIR=/home/devansh/bun/scripts
++ SCRIPT_DIR=/home/devansh/bun/scripts
++ export BUN_BASE_DIR=/home/devansh/bun
++ BUN_BASE_DIR=/home/devansh/bun
++ export BUN_DEPS_DIR=/home/devansh/bun/src/deps/
++ BUN_DEPS_DIR=/home/devansh/bun/src/deps/
++ export BUN_DEPS_OUT_DIR=/home/devansh/bun/src/deps/
++ BUN_DEPS_OUT_DIR=/home/devansh/bun/src/deps/
++ export CC=/usr/bin/clang-16
++ CC=/usr/bin/clang-16
++ export CXX=/usr/bin/clang++-16
++ CXX=/usr/bin/clang++-16
++ export AR=/usr/bin/ar
++ AR=/usr/bin/ar
++ export CPUS=12
++ CPUS=12
++ export CMAKE_CXX_COMPILER=/usr/bin/clang++-16
++ CMAKE_CXX_COMPILER=/usr/bin/clang++-16
++ export CMAKE_C_COMPILER=/usr/bin/clang-16
++ CMAKE_C_COMPILER=/usr/bin/clang-16
++ export 'CFLAGS=-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden'
++ CFLAGS='-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden'
++ export 'CXXFLAGS=-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden'
++ CXXFLAGS='-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden'
++ CMAKE_FLAGS=('-DCMAKE_C_COMPILER=/usr/bin/clang-16' '-DCMAKE_CXX_COMPILER=/usr/bin/clang++-16' '-DCMAKE_C_FLAGS=-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden' '-DCMAKE_CXX_FLAGS=-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden' '-DCMAKE_BUILD_TYPE=Release')
++ export CMAKE_FLAGS
+++ uname -s
++ [[ Linux == \D\a\r\w\i\n ]]
++ mkdir -p /home/devansh/bun/src/deps/
++ [[ /home/devansh/bun/scripts/env.sh == \/\h\o\m\e\/\d\e\v\a\n\s\h\/\b\u\n\/\s\c\r\i\p\t\s\/\b\u\i\l\d\-\m\i\m\a\l\l\o\c\-\d\e\b\u\g\.\s\h ]]
+ MIMALLOC_OVERRIDE_FLAG=
+ MIMALLOC_VALGRIND_ENABLED_FLAG=
+ cd /home/devansh/bun/src/deps//mimalloc
+ rm -rf CMakeCache.txt CMakeFiles
+ cmake -DCMAKE_C_COMPILER=/usr/bin/clang-16 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-16 '-DCMAKE_C_FLAGS=-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden' '-DCMAKE_CXX_FLAGS=-O3 -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden' -DCMAKE_BUILD_TYPE=Release . -DCMAKE_BUILD_TYPE=Debug -DMI_DEBUG=1 -DMI_SKIP_COLLECT_ON_EXIT=1 -DMI_BUILD_SHARED=OFF -DMI_BUILD_STATIC=ON -DMI_BUILD_TESTS=OFF -DMI_OSX_ZONE=OFF -DMI_OSX_INTERPOSE=OFF -DMI_TRACK_VALGRIND=ON -DMI_BUILD_OBJECT=ON -DMI_USE_CXX=ON -DMI_OVERRIDE=OFF -DMI_OSX_ZONE=OFF -GNinja
-- The C compiler identification is Clang 16.0.6
-- The CXX compiler identification is Clang 16.0.6
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang-16 - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-16 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- 
-- Looking for include files valgrind/valgrind.h, valgrind/memcheck.h
-- Looking for include files valgrind/valgrind.h, valgrind/memcheck.h - found
-- Compile with Valgrind support (MI_TRACK_VALGRIND=ON)
-- Skip collecting memory on program exit (MI_SKIP_COLLECT_ON_EXIT=ON)
-- Use the C++ compiler to compile (MI_USE_CXX=ON)
-- 
-- Library base name: mimalloc-valgrind-debug
-- Version          : 2.1
-- Build type       : debug
-- C++ Compiler     : /usr/bin/clang++-16
-- Compiler flags   : -Wno-deprecated;-Wall;-Wextra;-Wno-unknown-pragmas;-fvisibility=hidden;-Wpedantic;-Wno-static-in-inline;-ftls-model=initial-exec
-- Compiler defines : MI_TRACK_VALGRIND=1;MI_SKIP_COLLECT_ON_EXIT=1
-- Link libraries   : /usr/lib/x86_64-linux-gnu/libpthread.a;/usr/lib/x86_64-linux-gnu/librt.a
-- Build targets    : static;object
-- 
-- Configuring done (0.9s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    MI_DEBUG


-- Build files have been written to: /home/devansh/bun/src/deps/mimalloc
+ ninja
[17/17] Building CXX object CMakeFiles/mimalloc-obj.dir/src/static.c.o
+ cp libmimalloc-debug.a /home/devansh/bun/src/deps//libmimalloc-debug.a
cp: cannot stat 'libmimalloc-debug.a': No such file or directory
Failed to build mimalloc-debug
error: script "setup" exited with code 1
               
 [~/bun]
 ✘  devansh   3297_stream_writable  gedit CMakeLists.txt
```


or should I install the CMake version 3.22.1 as this is the minimum required version?

--- a/scripts/build-mimalloc-debug.sh
+++ b/scripts/build-mimalloc-debug.sh
@@ -27,5 +27,14 @@ cmake "${CMAKE_FLAGS[@]}" . \
 
 ninja
 
-cp libmimalloc-debug.a $BUN_DEPS_OUT_DIR/libmimalloc-debug.a
+if [ -f libmimalloc-valgrind-debug.a ];
+    file="libmimalloc-valgrind-debug.a"
+elif [ -f libmimalloc-debug.a ];
+    file="libmimalloc-debug.a"
+else
+    echo "Could not find libmimalloc-valgrind-debug.a or libmimalloc-debug.a"
+    exit 1
+fi
+
+cp $file $BUN_DEPS_OUT_DIR/libmimalloc-debug.a
 cp CMakeFiles/mimalloc-obj.dir/src/static.c.o $BUN_DEPS_OUT_DIR/libmimalloc-debug.o

--- a/scripts/build-mimalloc-debug.sh
+++ b/scripts/build-mimalloc-debug.sh
@@ -27,9 +27,9 @@ cmake "${CMAKE_FLAGS[@]}" . \
 
 ninja
 
-if [ -f libmimalloc-valgrind-debug.a ];
+if [ -f libmimalloc-valgrind-debug.a ]; then
     file="libmimalloc-valgrind-debug.a"
-elif [ -f libmimalloc-debug.a ];
+elif [ -f libmimalloc-debug.a ]; then
     file="libmimalloc-debug.a"
 else
     echo "Could not find libmimalloc-valgrind-debug.a or libmimalloc-debug.a"

